core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -0,0 +1,213 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the select skill viewer.
+ */
+
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { CategorizedSkills } from 'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
+import { SkillSummary } from 'core/templates/domain/skill/skill-summary.model';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { GroupedSkillSummaries } from 'pages/skill-editor-page/services/skill-editor-state.service';
+import { FilterForMatchingSubstringPipe } from 'filters/string-utility-filters/filter-for-matching-substring.pipe';
+import { ShortSkillSummary } from 'core/templates/domain/skill/ShortSkillSummaryObjectFactory';
+
+require('domain/utilities/url-interpolation.service.ts');
+
+@Component({
+  selector: 'skill-selector',
+  templateUrl: './skill-selector.component.html',
+})
+export class SkillSelectorComponent implements OnInit {
+  // If countOfSkillsToPrioritize > 0, then sortedSkillSummaries should
+  // have the initial 'countOfSkillsToPrioritize' entries of skills with
+  // the same priority.
+  @Input() sortedSkillSummaries: GroupedSkillSummaries;
+  @Input() selectedSkillId: string;
+  @Input() countOfSkillsToPrioritize: number;
+  @Input() categorizedSkills: CategorizedSkills;
+  @Input() untriagedSkillSummaries: SkillSummary[];
+  @Input() allowSkillsFromOtherTopics: boolean;
+  @Output() selectedSkillIdChanged: EventEmitter<string> = new EventEmitter();
+  currCategorizedSkills: CategorizedSkills = null;
+  selectedSkill = null;
+  skillFilterText = '';
+  topicFilterList = [];
+  subTopicFilterDict: { [topicName: string]:
+     { subTopicName: string; checked: boolean }[] } = {};
+  intialSubTopicFilterDict = {};
+
+  private filterForMatchingSubtringPipe: FilterForMatchingSubstringPipe = (
+    new FilterForMatchingSubstringPipe()
+  );
+
+  constructor() {}
+
+  ngOnInit(): void {
+    this.currCategorizedSkills = this.categorizedSkills;
+    for (let topicName in this.currCategorizedSkills) {
+      var topicNameDict = {
+        topicName: topicName,
+        checked: false
+      };
+      this.topicFilterList.push(topicNameDict);
+      var subTopics = this.currCategorizedSkills[topicName];
+      this.subTopicFilterDict[topicName] = [];
+      for (let subTopic in subTopics) {
+        var subTopicNameDict = {
+          subTopicName: subTopic,
+          checked: false
+        };
+        this.subTopicFilterDict[topicName].push(subTopicNameDict);
+      }
+    }
+    this.intialSubTopicFilterDict = angular.copy(this.subTopicFilterDict);
+  }
+
+  checkIfEmpty(skills: string[]): boolean {
```

Comment:
Maybe change type to `SkillSummary[]`

Reply:
Hm I think this can either be `SkillSummary[]` or `ShortSkillSummary[]` based on which list is passed in (categorized skills or untriaged skills). Changing to `any[]` type for now but I don't think this is the best option either since it's not safe to use `any` (though I don't think `string` is any better).